### PR TITLE
Introduced env variables

### PR DESCRIPTION
**Pull request by jendab**

### CODE REVIEW

File: awsenergylabelercli/awsenergylabelercli.py [hunk 1]
```diff
@@ -106,17 +112,18 @@ def get_arguments():
...
                         '-r',
-                        default=None,
+                        default=os.environ.get('REGION'),
                         type=security_hub_region,
                         required=False,
                         help='The home AWS region, default is None')
     parser.add_argument('--frameworks',
                         '-f',
-                        default=["aws-foundational-security-best-practices"],
+                        default=os.environ.get('FRAMEWORKS', ["aws-foundational-security-best-practices"]),
```
Comment (costastf) [inline awsenergylabelercli/awsenergylabelercli.py:126]: If the default argument here is a list then we should cast the env variable in a list. Env vars are always only string so some transformation would be required here.

File: awsenergylabelercli/awsenergylabelercli.py [hunk 2]
```diff
@@ -126,36 +133,37 @@ def get_arguments():
     account_list.add_argument('--allowed-account-ids',
                               '-a',
                               nargs='*',
-                              default=None,
+                              default=os.environ.get('ALLOWED_ACCOUNT_IDS'),
```
Comment (costastf) [inline awsenergylabelercli/awsenergylabelercli.py:136]: This is also a list and would require casting.

**Review by costastf — COMMENTED** [review]
Please check all defaultly set environment variables that are not strings and make sure that we cast accordingly.

File: awsenergylabelercli/awsenergylabelercli.py (hunk 2 above)
Comment (jendab) [reply awsenergylabelercli/awsenergylabelercli.py:139]: This is covered with lines 198-204, which utilizes validators from the library (which also already check mutually exclusivity)

File: awsenergylabelercli/awsenergylabelercli.py [hunk 3]
```diff
@@ -82,22 +89,24 @@ def get_arguments():
...
                         help='The location of the logging config json file',
-                        default='')
+                        default=os.environ.get('LOG_CONFIG', ''))
     parser.add_argument('--log-level',
                         '-L',
                         help='Provide the log level. Defaults to info.',
                         dest='log_level',
                         action='store',
-                        default='info',
+                        default=os.environ.get('LOG_LEVEL', 'info'),
```
Comment (costastf) [inline awsenergylabelercli/awsenergylabelercli.py:98]: LOG_LEVEL is a bit too generic to assume it is targeting this specific software. If we want to support env vars for this option I would suggest something a bit more opinionated to make sure we are not picking up irrelevant settings intended for other software.

File: awsenergylabelercli/awsenergylabelercli.py [hunk 4]
```diff
@@ -82,22 +89,24 @@ def get_arguments():
...
                         choices=['debug',
                                  'info',
                                  'warning',
                                  'error',
                                  'critical'])
-    scope = parser.add_mutually_exclusive_group(required=True)
+    scope = parser.add_mutually_exclusive_group(required=not (os.environ.get('LANDING_ZONE_NAME') or
```
Comment (costastf) [inline awsenergylabelercli/awsenergylabelercli.py:104]: Not sure exactly what this is supposed to be doing?

Comment (costastf) [reply awsenergylabelercli/awsenergylabelercli.py:104]: Do you think that dynamically changing the behavior of the tool based on the environment is the most elegant way to go for this?

**Pull request merged**